src/shared/sam/debugger/awsSamDebugger.ts

```diff
@@ -177,7 +178,16 @@ export class SamDebugConfigProvider implements vscode.DebugConfigurationProvider
                     }
                     const resources = getResourcesFromTemplateDatum(templateDatum)
                     for (const resourceKey of resources.keys()) {
-                        const runtimeName = resources.get(resourceKey)?.Properties?.Runtime
+                        const resource = resources.get(resourceKey)
+                        // Skip it if there is not actually a resource or it's not a funciton
+                        if (
+                            !resource ||
+                            (resource.Type !== CloudFormation.LAMBDA_FUNCTION_TYPE &&
+                                resource.Type !== CloudFormation.SERVERLESS_FUNCTION_TYPE)
+                        ) {
+                            continue
+                        }
+                        const runtimeName = resource?.Properties?.Runtime
                         configs.push(
                             createTemplateAwsSamDebugConfig(
```

Comment:
should this be skipped for `value.Type === 'Api'` ?

Reply:
Yes because we don't get enough information from it, if we want to expand what we support in the future we can do so

Reply:
but it doesn't look like it is skipped currently.

Reply:
`AWS::Serverless::Api` and Event:Api are different, `AWS::Serverless::Api` has no information about the lambda so we can't use it to make a complete run configuration